server/routes/referrals/referralFormPresenter.ts

```diff
@@ -73,41 +87,83 @@ export default class ReferralFormPresenter {
         type: 'single',
         title: 'Check your answers',
         number: '4',
-        status: this.canSubmitReferral ? ReferralFormStatus.NotStarted : ReferralFormStatus.CannotStartYet,
+        status: checkAnswersSectionFormStatus,
         tasks: [
           {
             title: 'Check your answers',
-            url: this.canSubmitReferral ? 'check-answers' : null,
+            url: checkAnswersSectionFormStatus !== ReferralFormStatus.CannotStartYet ? 'check-answers' : null,
           },
         ],
       },
     ]
   }
 
-  private determineInterventionDetailsSectionStatus(): ReferralFormStatus {
-    const hasCompletedSection = [
+  private referralFormStatus(
+    draftReferralValues: DraftReferralValues,
+    previousFormStatus: ReferralFormStatus = ReferralFormStatus.Completed
+  ): ReferralFormStatus {
+    if (previousFormStatus !== ReferralFormStatus.Completed) {
+      return ReferralFormStatus.CannotStartYet
+    }
+    if (
+      draftReferralValues.every(field => {
+        if (field === null) {
+          return false
+        }
+        return Array.isArray(field) ? field.length !== 0 : true
+      })
+    ) {
+      return ReferralFormStatus.Completed
+    }
+    return ReferralFormStatus.NotStarted
+  }
+
+  private get requiredServiceUserFormValues(): DraftReferralValues {
+    return [
+      this.referral.serviceUser.crn,
+      this.referral.serviceUser.title,
+      this.referral.serviceUser.firstName,
+      this.referral.serviceUser.lastName,
+      this.referral.serviceUser.dateOfBirth,
+      this.referral.serviceUser.gender,
+      this.referral.serviceUser.ethnicity,
+      this.referral.serviceUser.preferredLanguage,
+      this.referral.serviceUser.religionOrBelief,
+      this.referral.needsInterpreter,
+      this.referral.hasAdditionalResponsibilities,
+    ]
+  }
+
+  private get requiredInterventionFormValues(): DraftReferralValues {
+    return [
       this.referral.relevantSentenceId,
       this.referral.desiredOutcomesIds,
       this.referral.complexityLevelId,
       this.referral.completionDeadline,
       this.referral.usingRarDays,
-    ].every(field => field !== null)
+    ]
+  }
 
-    return hasCompletedSection ? ReferralFormStatus.Completed : ReferralFormStatus.NotStarted
+  /* TODO: Page form values need to be defined */
+  private get requiredResponsibleOfficerFormValues(): DraftReferralValues {
+    return []
   }
 
-  private get canSubmitReferral(): boolean {
-    return this.determineInterventionDetailsSectionStatus() === ReferralFormStatus.Completed
+  /* TODO: Page form values need to be defined */
```

Comment:
If you haven't already, would you mind adding a ticket to Jira for these 2 TODOs?

Reply:
Yes. I will create those tickets.